src/scripts/ping-okta-server.ts

```diff
@@ -0,0 +1,152 @@
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import { RootCommand } from '..';
+import * as Console from 'fp-ts/lib/Console';
+import { pipe } from 'fp-ts/lib/function';
+import { Argv } from 'yargs';
+import { oktaAPIError } from '../schema';
+import {
+  oktaReadOnlyClient,
+  validateCredentials,
+} from './services/client-service';
+import { pingOktaServer } from './services/validation-service';
+
+/**
+ * Pings the okta server to see if it is up and running.
+ * @param clientId - the client id of the okta application.
+ * @param organisationUrl - the url of the okta organisation.
+ * @returns a TaskEither that resolves to a string if the okta server is up and running, otherwise an error message.
+ */
+export const validateOktaServerIsRunning = (
+  clientId: string,
+  organisationUrl: string
+): TE.TaskEither<Error, 'Okta server is up and running.'> =>
+  pipe(
+    Console.info('Pinging okta server...'),
+    TE.rightIO,
+    // eslint-disable-next-line functional/functional-parameters
+    TE.chain(() =>
+      TE.chain(
+        ({
+          organisationUrl,
+          clientId,
+        }: {
+          readonly organisationUrl: string;
+          readonly clientId: string;
+        }) => pingOktaServer(clientId, organisationUrl)
+      )(TE.right({ organisationUrl, clientId }))
+    ),
+    // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+    TE.chain((response) => {
+      // eslint-disable-next-line functional/no-conditional-statement
+      if (response.status >= 500) {
+        return TE.left(
+          new Error('Server error. Please wait and try again later.', {
+            cause: response,
+          })
+        );
+      }
+      // eslint-disable-next-line functional/no-conditional-statement
+      if (response.status >= 400 && response.status < 500) {
+        return TE.left(
+          new Error(
+            `Client error. Please check your client id [${clientId}] and the URL of your organisation [${organisationUrl}].`,
+            {
+              cause: response,
+            }
+          )
+        );
+      }
+      // eslint-disable-next-line functional/no-conditional-statement
+      if (response.status >= 200 && response.status < 300) {
+        return TE.right('Okta server is up and running.');
+      }
+      return TE.left(
+        new Error('Unexpected response from pinging okta server.', {
+          cause: response,
+        })
+      );
+    })
+  );
+
+/**
+ * Maps errors relating to credentials to a more specific error message.
+ * @param result - the result of a service
+ * @returns the result of the service if it was successful, otherwise a refined error in consideration to credentials.
+ */
+export const mapCredentialErrors = <T>(
```

Comment:
**Issue:** consider moving this inside `validateCredentials`. I can't see a case where you would **not** want to do this after checking the credentials.